scoring/score.py

```diff
@@ -1,14 +1,54 @@
+import collections
+
+EXPECTED_TOKENS = {'S': 8, 'G': 8}
+VALID_TOKENS = set(EXPECTED_TOKENS.keys())
+
+
 class InvalidScoresheetException(Exception):
     pass
 
 
 class Scorer(object):
     def __init__(self, teams_data, arena_data):
         self._teams_data = teams_data
-        self._arena_data = arena_data
+        self._zone_contents = {
+            k: collections.Counter(zone['tokens'].replace(' ', ''))
+            for k, zone in arena_data.items()
+        }
 
     def calculate_scores(self):
-        raise NotImplementedError()
+        def zone_score(zone_data):
+            if not zone_data:
+                return 0
+            return  3 * sum(zone_data.values()) / len(zone_data)
```

Comment:
I don't think this is quite correct.

> If tokens of both colours are present in a scoring zone,  the value of each token in that scoring zone is reduced from 3 to 1.

This looks to half the points at the end, making the tokens worth 1.5 points rather than 1 point.

eg https://github.com/srobo/sr2020-comp/pull/1/files#diff-e58af9df2705fcda0714a31abe6a7985R141 should be 2 points rather than 3.

Reply:
Ah, good spot. This is a failing in my recollection of the rules. I even had the rulebook open when I was doing this! :man_facepalming:

Reply:
c44f3dc